substratools/task_resources.py

```diff
@@ -2,72 +2,103 @@
 
-    # TODO: we may want to pass List[str] instead
     _values: Dict[str, List[str]]
 
     def __init__(self, argstr: str) -> None:
         """Create outputs from argument string
 
         Argument is expected to be a JSON array like:
         [
-            {"id": "local", "value": "/sandbox/output/model/uuid"},
+            {"id": "local", "value": "/sandbox/output/model/uuid", "multiple": False},
             {"id": "shared", ...}
         ]
         """
         self._values = {}
-
         argument_list = json.loads(argstr)
 
+        _check_resources(argument_list)
+
         for item in argument_list:
-            self._values.setdefault(item[_RESOURCE_ID], list()).append(item[_RESOURCE_VALUE])
+            self._values.setdefault(
+                item[_RESOURCE_ID], {_RESOURCE_VALUE: [], _RESOURCE_MULTIPLE: item[_RESOURCE_MULTIPLE]}
+            )
+            self._values[item[_RESOURCE_ID]][_RESOURCE_VALUE].append(item[_RESOURCE_VALUE])
 
-    def get_value(self, key: str) -> str:
-        """Return a value corresponding to the given key.
-        It will raise if there is no input for given key or if there are multiple values"""
-        val = self._values[key]
-        if len(val) > 1:
-            raise exceptions.InvalidInputOutputsError("there are more than one path")
+        self.opener_path = self.get_value(TASK_IO_OPENER)
+        self.input_data_folder_paths = self.get_value(TASK_IO_DATASAMPLES)
+        self.chainkeys_path = self.get_value(TASK_IO_CHAINKEYS)
 
-        return val[0]
+    def get_value(self, key: str) -> Optional[Union[List[str], str]]:
+        """Returns the value for a given key. Won't raise if there is no matching resource.
+        Will raise if there is a mismatch between the given multiplicity and the number of returned
+        elements.
 
-    def get_values(self, key: str) -> List[str]:
-        """get_values will return the list of str corresponding to the given key.
-        It will raise if key does not exist.
+        If multiple is True, will return a list else will return a single value
         """
-        return self._values[key]
-
-    def get_optional_value(self, key: str) -> Optional[str]:
-        """Return value for given key, won't raise if there is no matching resource.
-        Will raise if there are more than one value."""
         if key not in self._values:
             return None
-        val = self._values[key]
 
-        if len(val) > 1:
-            raise exceptions.InvalidInputOutputsError("there are more than one path")
+        val = self._values[key][_RESOURCE_VALUE]
+        multiple = self._values[key][_RESOURCE_MULTIPLE]
+
+        if not multiple and len(val) > 1:
+            raise exceptions.InvalidInputOutputsError(
+                f"There is more than one path for the non multiple resource {key}"
+            )
+
+        if multiple:
+            return val
 
         return val[0]
```

Comment:
what if val is an empty List ?

Reply:
can't be,  there are check on TaskResources init